devtools/maven/src/main/java/io/quarkus/maven/DevMojo.java

```diff
@@ -419,24 +420,40 @@ private void executeIfConfigured(String pluginGroupId, String pluginArtifactId,
                         version(plugin.getVersion()),
                         plugin.getDependencies()),
                 goal(goal),
-                getPluginConfig(plugin),
+                getPluginConfig(plugin, goal),
                 executionEnvironment(
                         project,
                         session,
                         pluginManager));
     }
 
-    private Xpp3Dom getPluginConfig(Plugin plugin) {
-        Xpp3Dom configuration = configuration();
-        Xpp3Dom pluginConfiguration = (Xpp3Dom) plugin.getConfiguration();
-        if (pluginConfiguration != null) {
-            //Filter out `test*` configurations
-            for (Xpp3Dom child : pluginConfiguration.getChildren()) {
+    private Xpp3Dom getPluginConfig(Plugin plugin, String goal) {
+        final Xpp3Dom configuration = configuration();
```

Comment:
To avoid confusion I would move this to the bottom of this method and rename it to `filteredConfig`.
I would also rename `dominantConfig` to `mergedConfig` or similar, but your mileage may vary.

Reply:
Ok, thanks, adjusted.